### DISCUSSION TITLE
Guide to Rust + Soroban contract testing

### DISCUSSION THREAD

**Discussion by wildework** [stellar/sorobanathon#63] [General]
I went through the original [Testing](https://soroban.stellar.org/docs/tutorials/testing) tutorial and I wanted to create something more basic and accessible. So I designed a new guide around a `Calculator` structure that has one function `add`. This allows for a direct comparison between Rust world and Soroban world, as well as keeps the number of concepts and types to minimum, which keeps the focus on Testing.

[sorobandev.com/guides/testing](https://sorobandev.com/guides/testing)

I plan on doing a bunch of these guides and working on the website itself to make it a great learning experience. Should I continue starting discussions for each guide I create (so far it's been roughly a day each)? Thanks!

**Comment by achallagundla** [comment]
YES 100% please continue to keep these coming. We will review these soon and get back to you!

**Comment by achallagundla** [comment]
Heya @wildework  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.
